scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,367 @@
+        raise ValueError('`intervals` must not contain nan.')
+    if np.isinf(intervals).all(axis=1).any():
+        raise ValueError('In each row in `intervals`, both values must not'
+                         ' be infinite.')
+    if (intervals[:, 0] > intervals[:, 1]).any():
+        raise ValueError('In each row of `intervals`, the left value must not'
+                         ' exceed the right value.')
+
+    uncensored = intervals[:, 0] == intervals[:, 1]
+    left_censored = np.isinf(intervals[:, 0])
+    right_censored = np.isinf(intervals[:, 1])
+    interval_censored = np.isfinite(intervals).all(axis=1) & ~uncensored
+
+    x2 = intervals[uncensored, 0]
+    left2 = intervals[left_censored, 1]
+    right2 = intervals[right_censored, 0]
+    intervals2 = intervals[interval_censored]
+
+    return x2, left2, right2, intervals2
+
+
+def _validate_x_censored(x, censored):
+    x = np.asarray(x)
+    if x.ndim != 1:
+        raise ValueError('`x` must be one-dimensional.')
+    censored = np.asarray(censored)
+    if censored.ndim != 1:
+        raise ValueError('`censored` must be one-dimensional.')
+    if (~np.isfinite(x)).any():
+        raise ValueError('`x` must not contain nan or inf.')
+    if censored.size != x.size:
+        raise ValueError('`x` and `censored` must have the same length.')
+    return x, censored.astype(bool)
+
+
+class CensoredData:
+    """
+    Instances of this class represent censored data.
+
+    Instances may be passed to the ``fit`` method of continuous
+    univariate SciPy distributions for maximum likelihood estimation.
+    The *only* method of the univariate continuous distributions that
+    understands `CensoredData` is the ``fit`` method.  An instance of
+    `CensoredData` can not be passed to methods such as ``pdf`` and
+    ``cdf``.
+
+    An observation is said to be *censored* when the precise value is unknown,
+    but it has a known upper and/or lower bound.  The conventional terminology
+    is:
+
+    * left-censored: an observation is below a certain value but it is
+      unknown by how much.
+    * right-censored: an observation is above a certain value but it is
+      unknown by how much.
+    * interval-censored: an observation lies somewhere on an interval between
+      two values.
+
+    Left-, right-, and interval-censored data can be represented by
+    `CensoredData`.
+
+    For interval-censored data, pass the lower and upper limits of
+    the intervals as one-dimensional arrays to the `CensoredData`
+    constuctor.  By using the values `np.inf` and `-np.inf`, left-
+    and right-censored data may also be created with the constructor.
+
+    For convenience, the class methods ``left_censored`` and
+    ``right_censored`` are provided to create a `CensoredData`
+    instance from a single one-dimensional array of measurements
+    and a corresponding boolean array to indicate which measurements
+    are censored.
+
+    Parameters
+    ----------
+    x : array_like, 1D
```

Comment:
Because currently it's only accepted by `rv_continuous.fit`, which is not vectorized. The other functions it will be accepted by are not yet vectorized, either.